### PR TITLE
Add `radioField`

### PR DESCRIPTION
**Pull request by amitaibu**
| Bootstrap | Tailwind |
| -- | -- |
|  image | image |

- [x] Add Radio field
- [x] Theme for Bootstrap (default)
- [x] Theme for Tailwind
- [x] Write tests
- [x] Update guide

### CODE REVIEW

File: IHP/View/CSSFramework.hs [hunk 1]
```diff
@@ -856,6 +902,52 @@ tailwind = def
...
+                            autofocus={autofocus}
+                            {...additionalAttributes}
+                        />
+                        {label}
+                    </div>
+                |]
+                    where
+                        -- @todo: Need to find a way to guarantee that the id is unique and html valid.
+                        -- use `generateUniqueId`?
+                        optionId = fieldInputId <> "-" <> optionValue
```
Comment (amitaibu) [inline IHP/View/CSSFramework.hs:947]: @mpscholten what do you think?

File: IHP/View/Form.hs [hunk 2]
```diff
@@ -759,6 +759,86 @@ selectField field items = FormField
...
+        , cssFramework = ?formContext.cssFramework
+        , helpText = ""
+        , placeholder = ""
+        , required = False
+        , autofocus = False
+    }
+    where
+        fieldName = symbolVal field
+        FormContext { model } = ?formContext
+{-# INLINE radioField #-}
```
Comment (amitaibu) [inline IHP/View/Form.hs:840]: What's the purpose of `INLINE`?

File: IHP/View/Form.hs [hunk 3]
```diff
@@ -759,6 +759,86 @@ selectField field items = FormField
...
+-- > </form>
+--
+-- If you want a certain value to be preselected, set the value in the controller. For example, to have the first user be preselected in the above example:
+--
+-- > action NewProjectAction = do
+-- >     users <- query @User |> fetch
+-- >     let userId = headMay users |> maybe def (.id)
+-- >     let target = newRecord @Project |> set #userId userId
+-- >     render NewView { .. }
+radioField :: forall fieldName model item.
```
Comment (amitaibu) [inline IHP/View/Form.hs:803]: Should we rename it to `radiosField` to indicate there could be multiple?

File: IHP/View/CSSFramework.hs [hunk 4]
```diff
@@ -199,6 +201,47 @@ instance Default CSSFramework where
...
+                                autofocus={autofocus}
+                                {...additionalAttributes}
+                            />
+                            {label}
+                        </div>
+                    |]
+                        where
+                            -- @todo: Need to find a way to guarantee that the id is unique and html valid.
+                            -- use `generateUniqueId`?
+                            optionId = fieldInputId <> "-" <> optionValue
```
Comment (mpscholten) [inline IHP/View/CSSFramework.hs:242]: typically we use `_` instead of `-` for form Ids in IHP

File: IHP/View/Form.hs (hunk 3 above)
Comment (mpscholten) [reply IHP/View/Form.hs:803]: Seems technically correct, but I'd propably frist try `radioField` when typing the code, so maybe we should stick with it as it feels more in line with the other form helpers

File: IHP/View/Form.hs (hunk 2 above)
Comment (mpscholten) [reply IHP/View/Form.hs:840]: When the code is inlined, this makes it easier for GHC to optimize the other very generic code. E.g. the code `cs (lcfirst (getModelName @model) <> "_" <> cs fieldName)` can be simplified a lot when the `model` type variable is statically known (which is typically the case when the helper is inlined at the right place)

**Comment by mpscholten** [comment]
Thanks, this is great work so far 👍

File: IHP/View/Form.hs (hunk 2 above)
Comment (amitaibu) [reply IHP/View/Form.hs:840]: How do you know when to use it? Is there some rule of thumb?

File: IHP/View/CSSFramework.hs (hunk 4 above)
Comment (amitaibu) [reply IHP/View/CSSFramework.hs]: Fixed

**Comment by amitaibu** [comment]
@mpscholten fixed the `-` to `_`. Ready for review..

File: IHP/View/Form.hs (hunk 2 above)
Comment (mpscholten) [reply IHP/View/Form.hs:840]: There's no hard rule. I've discovered some of those places by dumping the GHC Core output and checking for potential places. GHC can also report some of those places using the `missed-specialisations` flag (it's enabled in IHP)

File: IHP/View/Form.hs [hunk 5]
```diff
@@ -759,6 +759,86 @@ selectField field items = FormField
         FormContext { model } = ?formContext
 {-# INLINE selectField #-}
 
+-- | Radio require you to pass a list of possible values to select. We use the same mechanism as for for `selectField`.
```
Comment (mpscholten) [inline IHP/View/Form.hs:762]:
```suggestion
-- | Radio require you to pass a list of possible values to select. We use the same mechanism as for for 'selectField'.
```

**Comment by mpscholten** [comment]
Thanks!

**Pull request merged**